### DISCUSSION TITLE
RuntimeError: ConvTranspose 3D is not supported on MPS

### DISCUSSION THREAD

**Discussion by ilikechopin** [ENHANCE-PET/MOOSE#165] [General]
Hi

I am quite new and in a basic level with everything here.
I downloaded moose and started the analysis of FDG PET/CT
but, I got the error message like this. 

"RuntimeError: ConvTranspose 3D is not supported on MPS"

I am currently using MacBook - M3! 
Any suggesting to solve this error? 

Thanks!

**Comment by LalithShiyam** [comment]
Hi @ilikechopin 

Please go through the read me for Mac installation. And please follow it to the last word. You need to install a specific branch of torch as mentioned in the readme

https://github.com/ENHANCE-PET/MOOSE?tab=readme-ov-file#for-macs-powered-by-apple-silicon-m-series-chips-with-mps-

**Reply by ilikechopin** [reply]
Thanks. I don't know what the problem was (as I followed the instruction on the website) 
But anyway, now it works! 
Merry Christmas!

**Reply by ilikechopin** [reply]
I have one more question. Would that be possible to select ROI instead of model?
for example, 

model: clin_ct_body  
There are 4 ROIs 
- 1:Legs, 2:Body, 3:Head, 4:Arms


If I need only "3:Head", can I run Moose for specific ROI segmentation of Head? instead of "clin_ct_body" (which would be faster)?

**Reply by LalithShiyam** [reply]
Hi @ilikechopin 

Great that it works. Unfortunately you can't choose individual rois, but you can do some easy post processing to extract the head. ChatGPT can whip you up a quick code.

Lalith

**Reply by ilikechopin** [reply]
Thanks! 

Arto

**Comment by LalithShiyam** [comment]
Hi there, No worries at all. Go to the path where you created the moose-env environment. Assuming you have created the environment in the Documents folder. Go there via the terminal  by typing: e.g cd ~Documents/Then you should “source” the moose-env to activate it. Think of this as a zoo “environment” where the moose lives. You need to enter the zoo, by typing the command: source moose-env/bin/activate. Once you do that you are inside the environment, and you can use moose as usual. Whenever you close the terminal, you need to do this again.Hope this helps. Cheers and Merry Christmas,Lalith  On Dec 24, 2024, at 9:10 AM, nutellaBear ***@***.***> wrote:﻿
Hi @ilikechopin
Great that it works. Unfortunately you can't choose individual rois, but you can do some easy post processing to extract the head. ChatGPT can whip you up a quick code.
Lalith
